server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java

```diff
@@ -132,31 +131,25 @@ private Mail toMail(EnqueuedItem enqueuedItem, MimeMessage mimeMessage) {
         return mail;
     }
 
-    private FluentFutureStream<EnqueuedItemWithSlicingContext> browseSlice(MailQueueName queueName, Slice slice) {
-        return FluentFutureStream.of(
+    private Mono<List<EnqueuedItemWithSlicingContext>> browseSlice(MailQueueName queueName, Slice slice) {
```

Comment:
`sort` is a bit scary because it actually collects without making it obvious, but I agree that the simpler data type is interesting, I'll change the code with your proposal